common/common.cpp

```diff
@@ -2361,10 +2361,22 @@ static bool llama_download_file(const std::string & url, const std::string & pat
         curl_easy_setopt(curl.get(), CURLOPT_HEADERFUNCTION, static_cast<CURLOPT_HEADERFUNCTION_PTR>(header_callback));
         curl_easy_setopt(curl.get(), CURLOPT_HEADERDATA, &headers);
 
-        CURLcode res = curl_easy_perform(curl.get());
-        if (res != CURLE_OK) {
-            fprintf(stderr, "%s: curl_easy_perform() failed: %s\n", __func__, curl_easy_strerror(res));
-            return false;
+        int head_request_attempts = 3;
+        int remaining_request_attempts = 3;
+        int retry_delay = 2;
+        while (remaining_request_attempts > 0){
+            CURLcode res = curl_easy_perform(curl.get());
+            if (res != CURLE_OK) {
+                int exponential_backoff_delay = std::pow(retry_delay, (head_request_attempts - remaining_request_attempts)) * 1000;
```

Comment:
Then use the defined:

```suggestion
                int exponential_backoff_delay = std::pow(CURL_RETRY_DELAY_SECONDS, (CURL_MAX_RETRY - remaining_request_attempts)) * 1000;
```